### PR TITLE
Stop RabbitMQ on each test class

### PR DESCRIPTION
**Pull request by aduprat**

### CODE REVIEW

File: backends-common/rabbitmq/src/test/java/org/apache/james/backend/rabbitmq/DockerRabbitMQTestRule.java [hunk 1]
```diff
@@ -18,24 +18,24 @@
...
     private DockerRabbitMQ dockerRabbitMQ;
 
     @Override
-    public Statement apply(Statement base, Description description) {
-        return new Statement() {
-            @Override
-            public void evaluate() throws Throwable {
-                dockerRabbitMQ = DockerRabbitMQ.withoutCookie();
-                dockerRabbitMQ.start();
-                base.evaluate();
```
Comment (rouazana) [inline backends-common/rabbitmq/src/test/java/org/apache/james/backend/rabbitmq/DockerRabbitMQTestRule.java]: this line was useless?

Comment (aduprat) [reply backends-common/rabbitmq/src/test/java/org/apache/james/backend/rabbitmq/DockerRabbitMQTestRule.java]: It's in the abstract class.

**Comment by trantienduchn** [comment]
What is the problem of keeping docker rabbit across test classes? I though we can reuse docker in this case and no need to bootstrap new docker container.

**Comment by aduprat** [comment]
Build took 2h40min

**Comment by aduprat** [comment]
> What is the problem of keeping docker rabbit across test classes? I though we can reuse docker in this case and no need to bootstrap new docker container.

Other docker related test rules are using a singleton for the container.
This means that every tests classes are using the same container, which may leads to other issues...

In this case, the container is recreated for each test class and it was not stop.
When playing test on a full project you can easily have more than 20 RabbitMQ servers running, which is not great for the performances at least it's a kind of memory leak.

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**